### DISCUSSION TITLE
Latent time plot not affected by root change

### DISCUSSION THREAD

**Discussion by akramdi** [theislab/scvelo#466] [Q&A]
Hello, 

In an effort to improve root detection in my data, I followed the solution suggested in this issue  #223 

I can see that this improves root detection as it makes it coincide with what I'm expecting biologically. However, the latent time plot does not seem to take the new root into account..

- Initial analysis: 

![scvelo_PDX63_MEStoNOR_J4_adjustedFilters_rmRibo_dynamical-mode_noGeneSelection_nei-30_ntopGenes-2000 lentent-time 0](https://user-images.githubusercontent.com/8793228/104954845-61db6980-59c9-11eb-9c74-a758466f4020.png)

- Improve root detection: 

```
adata = scv.read(outObj)
scv.tl.velocity_graph(adata, mode_neighbors='connectivities')
scv.tl.terminal_states(adata)
scv.tl.latent_time(adata)
outPNG="{}.latent-time.2.png".format(experimentTAG)
scv.pl.scatter(adata, color=['root_cells', 'end_points','latent_time',], color_map='viridis', size=80, dpi = 300,  figsize=(7,5), save = outPNG, show=False)

```
![scvelo_PDX63_MEStoNOR_J4_adjustedFilters_rmRibo_dynamical-mode_noGeneSelection_nei-30_ntopGenes-2000 latent-time 2](https://user-images.githubusercontent.com/8793228/104954719-10cb7580-59c9-11eb-8cdd-5b19f9be6576.png)

It looks identical. I even tried setting up a random index as root cell and still no effect on the latent time plot:
```

adata = scv.read(outObj)
scv.tl.velocity_graph(adata, mode_neighbors='connectivities')
scv.tl.terminal_states(adata)
scv.tl.latent_time(adata, root_key=5)
outPNG="{}.latent-time.random.png".format(experimentTAG)
scv.pl.scatter(adata, color=['root_cells', 'end_points','latent_time',], color_map='viridis', size=80, dpi = 300,  figsize=(7,5), save = outPNG, show=False)

```

![scvelo_PDX63_MEStoNOR_J4_adjustedFilters_rmRibo_dynamical-mode_noGeneSelection_nei-30_ntopGenes-2000 latent-time random](https://user-images.githubusercontent.com/8793228/104954987-ae26a980-59c9-11eb-8696-ef8c35ce6360.png)

Something is clearly off. Could you help me with this ?
Many thanks in advance.

Best,
Amira

**Comment by WeilerP** [comment]
Did you check numerically that the latent time did not change? The two scatter plots colored by latent time look different to me (the values in the purple part of the lower cluster are elevated in the improved root detection).
